examples/contentful-simple/lib/makeswift/components/blog-post/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { MakeswiftComponent } from '@makeswift/runtime/next'
+import { getSiteVersion } from '@makeswift/runtime/next/server'
+
+import { GetBlogsDocument, GetBlogsQuery } from '@/generated/contentful'
+import { client } from '@/lib/contentful/client'
+import { BlogPostProvider } from '@/lib/contentful/provider'
+import { client as MakeswiftClient } from '@/lib/makeswift/client'
+
+import { COMPONENT_TYPE } from './register'
+
+type Props = {
+  snapshotId?: string
+  label?: string
+  slug: string
+}
+
+export const BlogPost = async ({
+  snapshotId = 'site-header',
+  label = 'Blog Post',
+  slug,
+}: Props) => {
+  const snapshot = await MakeswiftClient.getComponentSnapshot(snapshotId, {
```

Comment:
As per my original comment, we should grab the `lib/makeswift/component.tsx` primitive from https://github.com/makeswift/makeswift/blob/main/examples/contentful/lib/makeswift/component.tsx and use that instead of `getComponentSnapshot` + `getSiteVersion` + `<MakeswiftComponent>`